### DISCUSSION TITLE
How to configure only one locator and one listening thread for communication using UDP

### DISCUSSION THREAD

**Discussion by Syamkamalnadhc** [eProsima/Fast-DDS#4616] [Q&A]
I have tried to run the helloworld example from eprosima FastDDS. I want to set a channel which is using only one listening thread and one unicast locator for a point-to-point communication. I tried disabling multicast traffic using the instructions from [documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/disabling_multicast.html), got 4 locators and two listening threads. Please give the settings

**Comment by JesusPoderoso** [comment]
Hi @Syamkamalnadhc, thanks for using Fast DDS.
If the [transport layer](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/transport.html) is not configured, Fast DDS uses by default the [builtin transports](https://fast-dds.docs.eprosima.com/en/latest/fastdds/rtps_layer/rtps_layer.html#managing-the-builtin-transports). Among other builtin transport configuration, the `DEFAULT` instanciates both UDPv4 and SHM transports. SHM transport has priority over the UDPv4 transport, meaning that SHM will always be used when possible. 
If you want to use a single transport as UDP, consider changing the builtin transport as explained in [this section of the documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/tcp/tcp_with_multicast_discovery.html#large-data-mode) (via environment variable, XML profile configuration or C++ code), and changing `LARGE_DATA` to `UDPv4`.

**Reply by Syamkamalnadhc** [reply]
Still, I am getting two receiver threads with single unicast locator with UDP. Can you help me to get idea about how the two threads are generating and is this thread used for participant and end point discovery? I could not identify why the second thread generating and who is generating the thread for what purpose.

**Comment by JesusPoderoso** [comment]
Hi @Syamkamalnadhc , can you provide the changes you've done in the example so we can reproduce your issue?
